exberry_adapter/bot/exberry_adapter_bot.py

```diff
@@ -249,25 +262,34 @@ async def handle_new_order_success(event):
         logging.info(f"Handling NewOrderSuccess")
 
         event_sid = event.cdata['sid']
+
+        if not event_sid in sid_to_order:
+            logging.error(f'Order {event.cdata} not found in adapter state')
+            return exercise(event.cid, 'Archive', {})
+
         order = sid_to_order.pop(event_sid)
 
         sid_is_cleared[event_sid] = order['isCleared']
 
         query = { 'order': order }
-
+        commands = [exercise(event.cid, 'Archive', {})]
         if order['isCleared']:
             logging.info(f"Acknowledging Cleared Order Success")
-            req_cid, _ = await client.find_one(MARKETPLACE.ClearedOrderRequest, query)
-
-            return [exercise(req_cid, 'ClearedOrderRequest_Ack', {
-                'orderId': event_sid
-            }), exercise(event.cid, 'Archive', {})]
+            try:
+                req_cid, _ = find_first(MARKETPLACE.ClearedOrderRequest, query)
+                commands.append(exercise(req_cid, 'ClearedOrderRequest_Ack', {
+                    'orderId': event_sid
+                }))
+            except: logging.error(f'Cleared Order request not found for {order}')
```

Comment:
do you know if the ordering of the `commands` list matters? 

before hand, this would've been `['ClearedOrderRequest_Ack', 'Archive']` but after the change it seems to look like `['Archive', 'ClearedOrderRequest_Ack']`

Is there a possibility of dazl archiving the contract before the request choice is exercised?

Reply:
I'm guessing that order does matter in a commands list for DAZL (though I'm 100% sure), but in this case the archive is on the incoming Exberry `NewOrderSuccess` contract and the choice is exercised on the `OrderRequest` contract.

Reply:
oh right!